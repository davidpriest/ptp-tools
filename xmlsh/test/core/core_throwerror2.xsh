# throw on error
# Test -e across scripts


if [ $# -gt 0 ] ; then
	echo in script arg count $#
	if [ $# -gt 1 ] ; then 
		shift 
		$0 "$@"
		exit 0;
	fi 
	[ $1 = "ignore" ] || echo Next command should fail 
	false
	
	[ $1 = "ignore" ] && exit 0 
	echo Fail SNH - after throw

fi

set -e 
try {
	
	$0 foo bar spam test
} catch E {
	echo Success caught $E
} 

# Test if calling script with condition doesnt throw
try {
	
	! $0 ignore
	echo Success 
} catch E {
	echo Fail - should not have caught error
} 