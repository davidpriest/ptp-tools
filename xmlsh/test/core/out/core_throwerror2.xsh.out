in script arg count 4
in script arg count 3
in script arg count 2
in script arg count 1
Next command should fail
Success caught 1
in script arg count 1
Success
